src/Services/DBusServer.vala

```diff
@@ -66,11 +66,28 @@ public class DBusServer : Object {
     public void uninstall (string component_id) throws Error {
         var client = AppCenterCore.Client.get_default ();
         var package = client.get_package_for_component_id (component_id);
+        var uninstall_confirm_dialog = new UninstallConfirmDialog (package);
+
         if (package == null) {
-            throw new IOError.FAILED ("Failed to find package for '%s' component ID".printf (component_id));
+            var error = new IOError.FAILED ("Failed to find package for '%s' component ID".printf (component_id));
+            new UninstallFailDialog (package, error).present ();
+            throw error;
+        }
+
+        if (uninstall_confirm_dialog.run () == Gtk.ResponseType.ACCEPT) {
+            package.uninstall.begin ((obj, res) => {
+                try {
+                    package.uninstall.end (res);
+                } catch (Error e) {
+                    // Disable error dialog for if user clicks cancel. Reason: Failed to obtain authentication
+                    if (e.code != 303) {
```

Comment:
I don't like this magic `303` number, can you do something more like:
```
if (e is GLib.IOError.CANCELLED) {
```

That might not be the right `CANCELLED` error type, but you get the idea.

Reply:
I am having issues with fixes it. The packagekit backend doesn't emit a GLib error here. It emits [Error](https://valadoc.org/packagekit-glib2/Pk.Error.html) class doesn't inherit GLib.Error. I think https://valadoc.org/packagekit-glib2/Pk.ErrorEnum.html used here to get the exact error. Need some guidance here on how to proceed

Reply:
@arshubham can't you do `if (e is Pk.ErrorEnum.NOT_AUTHORIZED) {` or whatever the correct error is here?

Reply:
No I get a `error: `Pk.ErrorEnum.NOT_AUTHORIZED' is not a type` for the above.